File: tvn-cosine.ai/tvn-cosine.ai/logic/fol/parsing/ast/FOLNode.cs
using tvn.cosine.collections.api;
using tvn.cosine.ai.logic.common;

namespace tvn.cosine.ai.logic.fol.parsing.ast
{
    public interface FOLNode : ParseTreeNode
    {
        string getSymbolicName(); 
        bool isCompound(); 
        ICollection<FOLNode> getArgs(); 
        object accept(FOLVisitor v, object arg); 
        FOLNode copy();
    }
}

File: tvn-cosine.ai/tvn-cosine.ai.v2/logic/basic/firstorder/parsing/ast/FOLNode.java
namespace aima.core.logic.basic.firstorder.parsing.ast;

using java.util.List;

using aima.core.logic.basic.firstorder.parsing.FOLVisitor;

/**
 * @author Ravi Mohan
 * @author Ciaran O'Reilly
 * @author Anurag Rai
 */
public interface FOLNode {
	String getSymbolicName();

	boolean isCompound();

	List<? extends FOLNode> getArgs();

	Object accept(FOLVisitor v, Object arg);

	FOLNode copy();
}
